rougelike_game/Game_working.py

```diff
@@ -1,76 +1,157 @@
-import subprocess as sp
+from Map_handler import *
 from Player import *
-
+import random
+import webbrowser
+import time
+import keyboard
+import subprocess as sp
 EMPTY_FIELD = '.'
 level = []
+e = 0
+again = ''
 
+b1 = 1
+b2 = 1
+level = handle_level(level)
+looking = '@'
 
+player1 = find_player(level, Player, looking)
 
-level_file = open('map.txt', 'r')
-level_str = level_file.read()
-rows = level_str.split('\n')
-
-for row in rows:
-    level.append(list(row))
-
-
-def draw_level(level):
-    for y in range(22):
-        for x in range(22):
-            print(level[y][x], end='')
-        print('')
-check_row = 0
-check_column = 0
-counter = 0
-position = 0
-print(check_row, check_column)
-
-for y in range(22):
-    for x in range(22):
-        if level[y][x] == '@':
-            print(y, x)
-            player1 = Player(y, x)
-            break
+looking = 'b'
 
+bot = find_player(level, Player, looking)
 
 
 
+#while True:  # making a loop
+ #   try:  # used try so that if user pressed other than the given key error will not be shown
+  #      if keyboard.is_pressed('q'):  # if key 'q' is pressed
+   #         print('You Pressed A Key!')
+    #        break  # finishing the loop
+    #except:
+     #   break  # if user pressed a key other than the given key the loop will break
 
 while True:
-
-    print(player1.y, player1.x)
     level[player1.y][player1.x] = '.'
-
-    command = input('> Please, write your command: ')
-    sp.call('clear', shell=True)
-    if command == 'exit':
-        print('Goodbye!')
+    level[bot.x][bot.y] = '.'
+    while True:
+        try:
+            if keyboard.is_pressed('e'):
+                print('EXIT y/n')
+                while True:
+                    try:
+                        if keyboard.is_pressed('y'):
+                            print('Goodbye!')
+                            b1 = 2
+                            b2 = 2
```

Comment:
you don't need to break using b1 and b2 param, you can just exit the program here using the `exit` function

Reply:
also, breaking out of the loop should be done using a parameter in the while instruction, for example instead of writing `while True` you can write `while b1 == 1` and event better you should use a boolean parameter like exitLoop = True and `while exitLoop == false`

Reply:
While using exit the command line is completly blank but the program is still running because when I want to exit the command line it gives me a notification that a program is running and when exiting I will kill it